### PR TITLE
feat: auto-scroll to the bottom of the conversation

### LINKED ISSUE #573: Enhancement: Auto scroll to bottom of conversation when opening
### Contact Details

_No response_

### What features would you like to see added?

In ChatGPT, when you open an existing conversation, it loads it with it already scrolled to the last message (no scroll animation). I'd like us to have something similar. I've played around with the code and got it to work like that. 

### More details

The change just requires calling `scrollToBottom()` when the Messages component is loaded.

```
  useEffect(() => {
    const timeoutId = setTimeout(() => {
      const { scrollTop, scrollHeight, clientHeight } = scrollableRef.current;
      const diff = Math.abs(scrollHeight - scrollTop);
      const percent = Math.abs(clientHeight - diff) / clientHeight;
      const hasScrollbar = scrollHeight > clientHeight && percent > 0.2;
      setShowScrollButton(hasScrollbar);
      // Scroll to bottom when conversation loads
      if (hasScrollbar) {
        scrollToBottom();
      }
    }, 650);
```

### Which components are impacted by your request?

UI

### Pictures

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by berry-13**
### on

image

### off

image

### issue:
~~- [ ] when you open a conversation with images, the conversation scrolls afterward~~

## Change Type

- [x] New feature (non-breaking change which adds functionality) 

## Testing

Check if the button works on different browser and with npm or docker.
Check if after refreshing the page it works

## Checklist

- [x] My code adheres to this project's style guidelines
- [x] I have performed a self-review of my own code
- [x] I have commented in any complex areas of my code
- [x] I have made pertinent documentation changes
- [x] My changes do not introduce new warnings
- [x] I have written tests demonstrating that my changes are effective or that my feature works
- [x] Local unit tests pass with my changes
- [x] Any changes dependent on mine have been merged and published in downstream modules.

### CODE REVIEW

**Comment by Berry-13** [comment]
@danny-avila, I'm not sure, but I believe the issue "when you open a conversation with images, the conversation scrolls afterward" can only be fixed by doing something like in #910.

**Comment by danny-avila** [comment]
> @danny-avila, I'm not sure, but I believe the issue "when you open a conversation with images, the conversation scrolls afterward" can only be fixed by doing something like in #910.

what do you mean by this?

**Comment by Berry-13** [comment]
> what do you mean by this?

I mean that when you open a conversation with images generated by stable diffusion or dall-e, first it load the images and then scroll down

**Comment by danny-avila** [comment]
> > what do you mean by this?
> 
> I mean that when you open a conversation with images generated by stable diffusion or dall-e, first it load the images and then scroll down

so the auto scroll you made doesn't happen in this case?

**Comment by danny-avila** [comment]
For context, got a reply on discord:
> when you open a conversation with images, for a fraction of a second you see the text or images at the beginning and after about half a second it scrolls down to the bottom of the conversation, this ONLY when there are images in the chat
so, yeah it's ready

File: client/src/store/settings.ts [hunk 1]
```diff
@@ -35,11 +35,31 @@ const showPopover = atom<boolean>({
   default: false,
 });
 
+const autoScroll = atom({
+  key: 'autoScroll',
+  default: localStorage.getItem('autoScroll') === 'true',
+  effects_UNSTABLE: [
```
Comment (danny-avila) [inline client/src/store/settings.ts]: based on this discussion, we can just use `effects` here unless it doesn't work for you: https://github.com/facebookexperimental/Recoil/discussions/1925

File: client/src/components/Messages/Message.tsx [hunk 2]
```diff
@@ -43,19 +44,26 @@ export default function Message({
...
+  const autoScroll = useRecoilValue(store.autoScroll);
 
   useEffect(() => {
     if (blinker && scrollToBottom && !abortScroll) {
       scrollToBottom();
     }
   }, [isSubmitting, blinker, text, scrollToBottom]);
 
+  useEffect(() => {
+    if (scrollToBottom && autoScroll) {
```
Comment (danny-avila) [inline client/src/components/Messages/Message.tsx]: Even though it shouldn't render in this case, I would just add a return condition if conversationId === 'new', return. This was a smart way of grabbing the convoId! 

I'm curious what kind of behavior there is when using Search, too, since it's pulling from params. Please test this. A search will change the params

Lastly, after thinking about this, you might be better off adding this to the <Messages/> component where Messages and we could probably move all the scrollToBottom() logic in Messages.

In other words, im not sure how important `message?.submitting` is.. if you are unsure what to do here, I can work on it after we merge this.

**Review by danny-avila — CHANGES_REQUESTED** [review]
